### ISSUE TITLE
explain() always returns error: None of these media types are available 

### ISSUE THREAD

**Issue by mozeryansky** [supabase/supabase#18933]
# Bug report

When I use `.explain()` on any of my queries I get this error:
```
{
  error: {
    code: 'PGRST107',
    details: null,
    hint: null,
    message: 'None of these media types are available: application/vnd.pgrst.plan+text; for="undefined"; options=analyze'
  },
  data: null,
  count: null,
  status: 415,
  statusText: 'Unsupported Media Type'
}
```

I also tried `.explain({ analyze: true })`, and I tried it on a few different queries.

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

`.explain()` isn't working

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:
```
await supabase
    .from('table')
    .select('*')
    .explain({ analyze: true })
```

## Expected behavior

Explain works

## System information

- Version of supabase-js: ^2.38.4
- Version of Node.js: v18.18.2

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/api/rest/debugging-performance#enabling-explain
Did you enable the mode needed for explain?
```sql
-- enable explain
alter role authenticator 
set pgrst.db_plan_enabled to 'true';

-- reload the config
notify pgrst, 'reload config';
```

**Comment by mozeryansky** [comment]
Yup, that was it! I was skimming https://supabase.com/docs/reference/javascript/explain, and didn't read the second half of the description there.

Any way to get the hint to be "see link ..." or "not enabled"? Either way, you can close this.

**Comment by GaryAustin1** [comment]
For the call to do that it would probably have to be done in PostgREST the REST API Supabase uses. 
You could certainly make an enhancement suggestion for them in discussions if you want.
